kart/resolve.py

```diff
@@ -42,41 +44,148 @@ def write_feature_to_dataset_entry(feature, dataset, repo):
+    dataset_part = rich_conflict.decoded_path[1]
+    if not single_path or dataset_part not in ("feature", "tile"):
         raise NotYetImplemented(
-            "Sorry, only feature conflicts can currently be resolved using --with-file"
+            "Sorry, only feature or tile conflicts can currently be resolved using --with=workingcopy"
         )
 
+    dataset_part = rich_conflict.decoded_path[1]
+    if dataset_part == "feature":
+        return _load_workingcopy_resolve_for_feature(rich_conflict)
+    elif dataset_part == "tile":
+        return _load_workingcopy_resolve_for_tile(rich_conflict)
+    else:
+        raise RuntimeError()
+
+
+def _load_workingcopy_resolve_for_feature(rich_conflict):
+    dataset = load_dataset(rich_conflict)
+    repo = dataset.repo
+    table_wc = repo.working_copy.tabular
+    pk = rich_conflict.decoded_path[2]
+    feature = (
+        table_wc.get_feature(dataset, pk, allow_schema_diff=False) if table_wc else None
+    )
+    if feature is None:
+        raise NotFound(
+            f"No feature found at {rich_conflict.label} - to resolve a conflict by deleting the feature, use --with=delete"
+        )
+    feature_path, feature_data = dataset.encode_feature(feature)
+    blob_id = repo.create_blob(feature_data)
+    return [pygit2.IndexEntry(feature_path, blob_id, pygit2.GIT_FILEMODE_BLOB)]
+
+
+def _load_workingcopy_resolve_for_tile(rich_conflict):
+    from kart.point_cloud.tilename_util import get_tile_path_pattern
+
+    dataset = load_dataset(rich_conflict)
+    repo = dataset.repo
+    workdir = repo.working_copy.workdir
+    tilename = rich_conflict.decoded_path[2]
+    matching_files = []
+    if workdir:
+        # Get a glob that roughly matches the tiles we are looking for.
+        matching_files = list((workdir.path / dataset.path).glob(f"**/{tilename}.*"))
+        # Narrow it down more exactly using get_tile_path_pattern which allows for a few different extensions.
+        filename_pattern = get_tile_path_pattern(tilename)
+        matching_files = [
+            p for p in matching_files if filename_pattern.fullmatch(p.name)
+        ]
+
+    if not matching_files:
+        raise NotFound(
+            f"No tile found at {rich_conflict.label} - to resolve a conflict by deleting the tile, use --with=delete"
+        )
+    if len(matching_files) > 1:
+        click.echo(
+            "Found multiple files in the working copy that could be intended as the resolution:",
+            err=True,
+        )
+        for file in matching_files:
+            click.echo(
+                os.path.relpath(file.resolve(), repo.workdir_path.resolve()), err=True
+            )
+        raise InvalidOperation("Couldn't resolve conflict using working copy")
+    return _load_file_resolve_for_tile(rich_conflict, matching_files[0])
 
-@click.command()
+
+CHOICE_ALIASES = {"working-copy": "workingcopy"}
+CONTEXT_SETTINGS = dict(token_normalize_func=lambda x: CHOICE_ALIASES.get(x, x))
```

Comment:
🙈 
does this thing normalise tokens before they're parsed? doesn't that mean that it could 'normalise' tokens that belong to a different option altogether? There's got to be a better way

Reply:
Removed it for now. Not sure what the better way is